google/cloud/security/inventory/pipelines/load_instance_group_managers_pipeline.py

```diff
@@ -0,0 +1,101 @@
+from google.cloud.security.common.util import log_util
+from google.cloud.security.common.util import parser
+from google.cloud.security.inventory import errors as inventory_errors
+from google.cloud.security.inventory.pipelines import base_pipeline
+
+LOGGER = log_util.get_logger(__name__)
+
+
+class LoadInstanceGroupManagersPipeline(base_pipeline.BasePipeline):
+    """Load compute instance group managers for all projects."""
+
+    RESOURCE_NAME = 'instance_group_managers'
+
+    def _transform(self, resource_from_api):
+        """Create an iterator of instance group managers to load into database.
+
+        Args:
+            resource_from_api: A dict of instance group managers, keyed by
+                project id, from GCP API.
+
+        Yields:
+            Iterator of instance group manager properties in a dict.
+        """
+        for (project_id, igms) in resource_from_api.iteritems():
+            for igm in igms:
+                yield {'project_id': project_id,
+                       'id': igm.get('id'),
+                       'creation_timestamp': parser.format_timestamp(
+                           igm.get('creationTimestamp'),
+                           self.MYSQL_DATETIME_FORMAT),
+                       'name': igm.get('name'),
+                       'description': igm.get('description'),
+                       'base_instance_name': igm.get('baseInstanceName'),
+                       'current_actions': parser.json_stringify(
+                           igm.get('currentActions', {})),
+                       'instance_group': igm.get('instanceGroup'),
+                       'instance_template': igm.get('instanceTemplate'),
+                       'named_ports': parser.json_stringify(
+                           igm.get('namedPorts', [])),
+                       'region': igm.get('region'),
+                       'target_pools': parser.json_stringify(
+                           igm.get('targetPools', [])),
+                       'target_size': igm.get('targetSize'),
+                       'zone': igm.get('zone')}
+
+    def _retrieve(self):
+        """Retrieve instance group managers from GCP.
+
+        Get all the projects in the current snapshot and retrieve the
+        compute instance group managers for each.
+
+        Returns:
+            A dict mapping projects with their instance group managers (list):
+            {project_id: [instance group managers]}
+        """
+        projects = proj_dao.ProjectDao().get_projects(self.cycle_timestamp)
+        igms = {}
+        for project in projects:
+            project_igms = []
+            try:
+                response = self.api_client.get_instance_group_managers(
+                    project.id)
+                for page in response:
+                    items = page.get('items', {})
+                    for group_igms in items.values():
+                        igm_instances = group_igms.get(
+                            'instanceGroupManagers', [])
```

Comment:
Hmmm.  Additional processing is done here by the caller of ```get_instance_group_managers()``` to get the actual instance group manager data, and aggregate the paged results.  What is your thought about having this done in the ```get_instance_group_managers()```, so that any other potential callers of this api won't have to do the additional processing?  (This will also apply to the other retrieve()'s in this PR.)

For example, you can take a look at the ```get_firewall_rules()``` in the same compute.py that returns only the firewall rules data, rather than just the raw gcp response.
https://github.com/GoogleCloudPlatform/forseti-security/blob/master/google/cloud/security/common/gcp_api/compute.py#L130
```python
        firewall_rules_api = self.service.firewalls()
        request = firewall_rules_api.list(project=project_id)

        paged_results = self._build_paged_result(request, firewall_rules_api,
                                                 self.rate_limiter)

        firewall_rules = []
        for page in paged_results:
            firewall_rules.extend(page.get('items', []))
        return firewall_rules
```

This way, the ```_retrieve()``` can be more simple as well:
```python
        firewall_rules_map = {}
        projects = self.dao.get_projects(self.cycle_timestamp)
        for project in projects:
            try:
                firewall_rules = self.api_client.get_firewall_rules(project.id)
                firewall_rules_map[project.id] = firewall_rules
            except api_errors.ApiExecutionError as e:
                LOGGER.error('Unable to get firewall rules for '
                             'project id: %s\n%s', project.id, e)
        return firewall_rules_map
```